drivers/serial/uart_cdns.c

```diff
@@ -209,9 +209,146 @@ static void uart_cdns_irq_handler(const struct device *dev)
 
 #endif
 
+#ifdef CONFIG_UART_USE_RUNTIME_CONFIGURE
+
+static enum uart_config_parity parity_from_cdns(enum csr_parity_val parity)
+{
+	switch (parity) {
+	case EVEN_PARITY_VAL:
+		return UART_CFG_PARITY_EVEN;
+	case ODD_PARITY_VAL:
+		return UART_CFG_PARITY_ODD;
+	case SPACE_PARITY_VAL:
+		return UART_CFG_PARITY_SPACE;
+	case MARK_PARITY_VAL:
+		return UART_CFG_PARITY_MARK;
+	case NO_PARITY_VAL:
+		return UART_CFG_PARITY_NONE;
+	}
+	return UART_CFG_PARITY_NONE;
+}
+
+static int32_t parity_to_cdns(enum uart_config_parity parity)
+{
+	switch (parity) {
+	case UART_CFG_PARITY_NONE:
+		return NO_PARITY_VAL;
+	case UART_CFG_PARITY_ODD:
+		return ODD_PARITY_VAL;
+	case UART_CFG_PARITY_EVEN:
+		return EVEN_PARITY_VAL;
+	case UART_CFG_PARITY_MARK:
+		return MARK_PARITY_VAL;
+	case UART_CFG_PARITY_SPACE:
+		return SPACE_PARITY_VAL;
+	}
+	return -1;
```

Comment:
Since this is used for bit manipulation later, please use unsigned integer with correct bits set instead.

Same for those below.